ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KtlintSuppressionRuleTest.kt

```diff
@@ -0,0 +1,798 @@
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive for which the target element is already annotated with @Suppress then add the ktlint suppression and sort all suppressions alphabetically`() {
+            val code =
+                """
+                /* ktlint-disable some-rule-set:some-rule-id */
+                @Suppress("zzz", "aaa")
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(1, 4, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive for which the target element is already annotated with @SuppressWarnings then add the ktlint suppression and sort all suppressions alphabetically`() {
+            val code =
+                """
+                /* ktlint-disable some-rule-set:some-rule-id */
+                @SuppressWarnings("aaa", "zzz")
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @SuppressWarnings("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(1, 4, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive for which the target element is already annotated with both @Suppress and @SuppressWarnings then add the ktlint suppression to the @Suppress`() {
+            val code =
+                """
+                /* ktlint-disable some-rule-set:some-rule-id */
+                @Suppress("aaa", "zzz")
+                @SuppressWarnings("bbb", "yyy")
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                @SuppressWarnings("bbb", "yyy")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(1, 4, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+    }
+
+    @Nested
+    inner class `Given a non-top level block comment with a ktlint-disable directive` {
+        @Test
+        fun `Given a ktlint-disable directive without rule-id`() {
+            val code =
+                """
+                fun foo() {
```

Comment:
what will be behaviour when `/* ktlint-disable */` is added to the `init { }` block?

Reply:
Given code below:
```
class Foo() {
    var foo: String
    var bar: String

    init {
        /* ktlint-disable standard:foo */
        foo = "foo"
    }

    init { // ktlint-disable standard:bar
        bar = "bar"
    }
}
```
this results in adding a @Suppress to both init blocks which is not valid code. I will add the test case and fix. Example above uncovered that after an initial fix, the @Suppress annotation on classes and functions is not inserted correctly in the AST resulting in two distinct annotation on top of the class. Need to do some refactoring there (not yet pushed).